jxl/src/render/internal.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+use crate::image::{DataTypeTag, ImageDataType};
+
+use super::{
+    RenderPipelineExtendStage, RenderPipelineInOutStage, RenderPipelineInPlaceStage,
+    RenderPipelineInputStage,
+};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum RenderPipelineStageType {
+    Input,
+    InPlace,
+    InOut,
+    Extend,
+}
+
+pub trait RenderPipelineStageInfo: super::simple_pipeline::RenderPipelineRunStage {
```

Comment:
I'm not sure if I understand the design correctly... How do you add additional render pipeline implementation, when `simple_pipeline::RenderPipelineRunStage` needs to be impl-ed for all stages? Doesn't it conflict with existing `RenderPipelineRunStage` impls?

Reply:
The idea would be that an additional implementation adds another constraint on `RenderPipelineStageInfo` for its own `RenderPipelineRunStage` trait, which is implemented (by the pipeline impl itself) on all the RenderPipelineExtendStage etc (which are just marker types to be used in `ActualStage::Type`).
Does that seem reasonable?

Reply:
Yeah that would work, I first thought the supertrait bound will not change. Though I think `#[cfg]`-ing render pipeline would be somewhat tedious, because every combination of cfg would need its own trait bound, but still it might not be an issue if there won't be so many implementations...

Reply:
I don't expect more than 2 implementations :-)
Also I did not come up with a better idea, so...